samples/shields/x_nucleo_cca02m1/src/main.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <misc/printk.h>
+
+#include <gpio.h>
+#include <led.h>
+
+#include <audio/dmic.h>
+
+/* uncomment if you want PCM output in ascii */
+/*#define PCM_OUTPUT_IN_ASCII		1  */
+
+#define STEREO	1
+
+#define AUDIO_FREQ		16000
+#define CHAN_SIZE		16
+#ifdef STEREO
+#define PCM_BLK_SIZE_MS		(((AUDIO_FREQ/1000) * sizeof(s16_t)) * 2)
+#else
+#define PCM_BLK_SIZE_MS		((AUDIO_FREQ/1000) * sizeof(s16_t))
+#endif
+
+#define NUM_MS		1000
+
+K_MEM_SLAB_DEFINE(rx_mem_slab, PCM_BLK_SIZE_MS, NUM_MS, 1);
+
+struct pcm_stream_cfg mic_streams = {
+	.pcm_rate = AUDIO_FREQ,
+	.pcm_width = CHAN_SIZE,
+	.block_size = PCM_BLK_SIZE_MS,
+	.mem_slab = &rx_mem_slab,
+};
+
+struct dmic_cfg cfg = {
+	.io = {
+		/* requesting a pdm freq around 2MHz */
+		.min_pdm_clk_freq = 1800000,
+		.max_pdm_clk_freq = 2500000,
+	},
+	.streams = &mic_streams,
+	.channel = {
+#ifdef STEREO
+		.req_num_chan = 2,
+#else
+		.req_num_chan = 1,
+#endif
+	},
+};
+
+void *rx_block[NUM_MS];
+size_t rx_size = PCM_BLK_SIZE_MS;
+
+#ifdef STEREO
+#include <stm32f4xx_ll_tim.h>
+
+/*
+ * MIC_CLK_x2:     PB4 (TIM3_CH1) - IN
+ * MIC_CLK_NUCLEO: PB5 (TIM3_CH2) - OUT
+ */
+static int AUDIO_IN_Timer_Init(void)
+{
+	TIM_TypeDef *tim3 = (TIM_TypeDef *)DT_TIM_STM32_3_BASE_ADDRESS;
```

Comment:
I don't think it is acceptable to propose in upstream a sample that bypass zephyr APIs.
If some HW functionality is required for the sample it should be accessed via a generic zephyr API.
If missing, such API should be proposed.

Reply:
I see your point. I'll try to elaborate more on this.